rmf_fleet_adapter/src/rmf_fleet_adapter/agv/FleetUpdateHandle.cpp

```diff
@@ -1009,10 +1010,18 @@ void FleetUpdateHandle::add_robot(
          fleet = shared_from_this()](
         rmf_traffic::schedule::Participant participant)
   {
-    const std::size_t charger_wp = fleet->_pimpl->get_nearest_charger(
-        start[0], fleet->_pimpl->charging_waypoints);
+    const auto charger_wp = fleet->_pimpl->get_nearest_charger(start[0]);
+    
+    if (!charger_wp.has_value())
+    {
+      throw std::runtime_error(
```

Comment:
Instead of throwing an error here, let me know if it is better to pass in a random waypoint as the charging waypoint but then warn users to call `RobotUpdateHandle::set_charger_waypoint()`.

Reply:
If we get `nullopt` for the nearest charger, then that implies there are no chargers in the whole graph. That is almost certainly a system integration oversight and it makes sense to throw an exception for it.